map/bookmark_manager.cpp

```diff
@@ -90,12 +94,90 @@ BookmarkManager::SharingResult GetFileForSharing(BookmarkManager::KMLDataCollect
     return {{categoryId}, BookmarkManager::SharingResult::Code::FileError, "Bookmarks file does not exist."};
 
   auto tmpFilePath = base::JoinPath(GetPlatform().TmpDir(), fileName + std::string{kKmzExtension});
-  if (!CreateZipFromPathDeflatedAndDefaultCompression(filePath, tmpFilePath))
+
+  if (!CreateZipFromFiles({filePath}, tmpFilePath))
     return {{categoryId}, BookmarkManager::SharingResult::Code::ArchiveError, "Could not create archive."};
 
   return {{categoryId}, std::move(tmpFilePath)};
 }
 
+std::string BuildIndexFile(std::vector<std::string> const & filesForIndex)
+{
+  std::string const filePath = base::JoinPath(GetPlatform().TmpDir(), "doc.kml");
+  FileWriter fileWriter(filePath);
+  std::string content = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+      "<kml xmlns=\"http://earth.google.com/kml/2.0\">\n"
+      "<Document>\n"
+      "<name>Organic Maps Bookmarks and Tracks</name>\n";
+  for (auto const & fileName : filesForIndex)
+  {
+    content.append("<NetworkLink><Link><name>");
+    content.append(base::GetNameFromFullPathWithoutExt(fileName));
+    content.append("</name><href>");
+    content.append(fileName);
+    content.append("</href></Link></NetworkLink>\n");
+  }
+  content.append("</Document>\n"
+             "</kml>");
+  fileWriter.Write(content.c_str(), content.length());
+  return filePath;
+}
+
+BookmarkManager::SharingResult ExportMultipleFiles(BookmarkManager::KMLDataCollectionPtr collection)
+{
+  auto const kmzFileName = "OrganicMapsBackup_" + std::to_string(base::GenerateYYMMDD(time(nullptr)));
+  auto kmzFilePath = base::JoinPath(GetPlatform().TmpDir(), kmzFileName + std::string{kKmzExtension});
+  auto const filesDir = "files/";
+  kml::GroupIdCollection categoriesIds;
+  for (auto const & kmlToExport : *collection)
+    categoriesIds.push_back(kmlToExport.second->m_categoryData.m_id);
+  std::vector<std::string> filesInArchive;
+  std::vector<std::string> pathsForArchive;
+
+  SCOPE_GUARD(deleteFileGuard, [&pathsForArchive]() {
```

Comment:
nit: Maybe we need a helper function that returns a scope guard to delete files or even directories (not in this PR).